<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>standard_image_widget</class>
 <widget class="QWidget" name="standard_image_widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>552</width>
    <height>1194</height>
   </rect>
  </property>
  <property name="maximumSize">
   <size>
    <width>552</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_2">
   <item row="0" column="0">
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QFrame" name="image_frame">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>532</width>
         <height>532</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>532</width>
         <height>532</height>
        </size>
       </property>
       <property name="acceptDrops">
        <bool>true</bool>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QWidget" name="controls_container" native="true">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QPushButton" name="export_button">
          <property name="cursor">
           <cursorShape>PointingHandCursor</cursorShape>
          </property>
          <property name="text">
           <string>Export</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="delete_button">
          <property name="cursor">
           <cursorShape>PointingHandCursor</cursorShape>
          </property>
          <property name="toolTip">
           <string>Permanently delete this image</string>
          </property>
          <property name="text">
           <string>Delete</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="to_canvas_button">
          <property name="cursor">
           <cursorShape>PointingHandCursor</cursorShape>
          </property>
          <property name="toolTip">
           <string>Send this image to the canvas</string>
          </property>
          <property name="text">
           <string>To Canvas</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QWidget" name="delete_confirmation" native="true">
       <layout class="QGridLayout" name="gridLayout">
        <item row="3" column="0" colspan="2">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Permanently delete?</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0" colspan="2">
         <widget class="QWidget" name="button_container" native="true">
          <layout class="QHBoxLayout" name="buttons">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
          </layout>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QPushButton" name="confirm_delete_button">
          <property name="cursor">
           <cursorShape>PointingHandCursor</cursorShape>
          </property>
          <property name="text">
           <string>Yes</string>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QPushButton" name="cancel_delete_button">
          <property name="cursor">
           <cursorShape>PointingHandCursor</cursorShape>
          </property>
          <property name="text">
           <string>No</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QLabel" name="label_2">
           <property name="font">
            <font>
             <pointsize>11</pointsize>
             <bold>false</bold>
            </font>
           </property>
           <property name="text">
            <string>Prompt</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPlainTextEdit" name="prompt">
           <property name="placeholderText">
            <string/>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="reset_prompt">
           <property name="text">
            <string>Reset</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QLabel" name="label_3">
           <property name="font">
            <font>
             <pointsize>11</pointsize>
             <bold>false</bold>
            </font>
           </property>
           <property name="text">
            <string>Negative Prompt</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPlainTextEdit" name="negative_prompt">
           <property name="placeholderText">
            <string/>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="reset_negative_prompt">
           <property name="text">
            <string>Reset</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QGroupBox" name="similar_groupbox">
       <property name="title">
        <string>Generate similar images</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_4">
        <property name="leftMargin">
         <number>4</number>
        </property>
        <property name="topMargin">
         <number>4</number>
        </property>
        <property name="rightMargin">
         <number>4</number>
        </property>
        <property name="bottomMargin">
         <number>4</number>
        </property>
        <item row="1" column="0" colspan="2">
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="SliderWidget" name="samples_widget" native="true">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>20</height>
             </size>
            </property>
            <property name="label_text" stdset="0">
             <string>Similarity</string>
            </property>
            <property name="slider_callback" stdset="0">
             <string notr="true">handle_value_change</string>
            </property>
            <property name="current_value" stdset="0">
             <number>100</number>
            </property>
            <property name="slider_maximum" stdset="0">
             <number>100</number>
            </property>
            <property name="spinbox_maximum" stdset="0">
             <double>1.000000000000000</double>
            </property>
            <property name="display_as_float" stdset="0">
             <bool>true</bool>
            </property>
            <property name="spinbox_single_step" stdset="0">
             <double>0.010000000000000</double>
            </property>
            <property name="spinbox_page_step" stdset="0">
             <double>0.010000000000000</double>
            </property>
            <property name="spinbox_minimum" stdset="0">
             <double>0.000000000000000</double>
            </property>
            <property name="slider_minimum" stdset="0">
             <number>0</number>
            </property>
            <property name="settings_property" stdset="0">
             <string>standard_image_widget_settings.image_similarity</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="generate_single_simillar_button">
            <property name="cursor">
             <cursorShape>PointingHandCursor</cursorShape>
            </property>
            <property name="toolTip">
             <string>Generate one variation</string>
            </property>
            <property name="text">
             <string>Single</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="generate_batch_similar_button">
            <property name="cursor">
             <cursorShape>PointingHandCursor</cursorShape>
            </property>
            <property name="toolTip">
             <string>Generate a batch of four variations</string>
            </property>
            <property name="text">
             <string>Batch</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="0" column="0" colspan="2">
         <widget class="QGroupBox" name="groupBox_2">
          <property name="font">
           <font>
            <pointsize>8</pointsize>
            <bold>false</bold>
           </font>
          </property>
          <property name="title">
           <string>Controlnet</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <widget class="QComboBox" name="controlnet"/>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="title">
        <string>Upscale</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="0" column="0">
         <widget class="QComboBox" name="upscale_model">
          <item>
           <property name="text">
            <string>RealESRGAN_x4plus</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>RealESRNet_x4plus</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>RealESRGAN_x4plus_anime_6B</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>RealESRGAN_x2plus</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>realesr-animevideov3</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>realesr-general-x4v3</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QPushButton" name="upscale_2x">
          <property name="cursor">
           <cursorShape>PointingHandCursor</cursorShape>
          </property>
          <property name="text">
           <string>2x Upscale</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QCheckBox" name="face_enhance">
          <property name="text">
           <string>Face enhance</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item row="7" column="0">
    <spacer name="verticalSpacer_2">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="4" column="0">
    <widget class="QTableWidget" name="tableWidget">
     <property name="styleSheet">
      <string notr="true"/>
     </property>
     <property name="autoScroll">
      <bool>false</bool>
     </property>
     <attribute name="horizontalHeaderVisible">
      <bool>false</bool>
     </attribute>
     <attribute name="verticalHeaderVisible">
      <bool>false</bool>
     </attribute>
     <column>
      <property name="text">
       <string>New Column</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>New Column</string>
      </property>
     </column>
    </widget>
   </item>
   <item row="5" column="0">
    <widget class="QWidget" name="batch_container" native="true">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>128</height>
      </size>
     </property>
     <property name="styleSheet">
      <string notr="true"/>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_2"/>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>SliderWidget</class>
   <extends>QWidget</extends>
   <header>airunner/widgets/slider/slider_widget</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>confirm_delete_button</sender>
   <signal>clicked()</signal>
   <receiver>standard_image_widget</receiver>
   <slot>confirm_delete()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>259</x>
     <y>552</y>
    </hint>
    <hint type="destinationlabel">
     <x>237</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cancel_delete_button</sender>
   <signal>clicked()</signal>
   <receiver>standard_image_widget</receiver>
   <slot>cancel_delete()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>532</x>
     <y>552</y>
    </hint>
    <hint type="destinationlabel">
     <x>368</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>to_canvas_button</sender>
   <signal>clicked()</signal>
   <receiver>standard_image_widget</receiver>
   <slot>image_to_canvas()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>541</x>
     <y>483</y>
    </hint>
    <hint type="destinationlabel">
     <x>85</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>delete_button</sender>
   <signal>clicked()</signal>
   <receiver>standard_image_widget</receiver>
   <slot>delete_image()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>326</x>
     <y>483</y>
    </hint>
    <hint type="destinationlabel">
     <x>310</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>generate_single_simillar_button</sender>
   <signal>clicked()</signal>
   <receiver>standard_image_widget</receiver>
   <slot>similar_image()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>824</y>
    </hint>
    <hint type="destinationlabel">
     <x>38</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>generate_batch_similar_button</sender>
   <signal>clicked()</signal>
   <receiver>standard_image_widget</receiver>
   <slot>similar_batch()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>489</x>
     <y>824</y>
    </hint>
    <hint type="destinationlabel">
     <x>154</x>
     <y>2</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>export_button</sender>
   <signal>clicked()</signal>
   <receiver>standard_image_widget</receiver>
   <slot>export_image()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>182</x>
     <y>483</y>
    </hint>
    <hint type="destinationlabel">
     <x>467</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>upscale_2x</sender>
   <signal>clicked()</signal>
   <receiver>standard_image_widget</receiver>
   <slot>upscale_2x_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>431</x>
     <y>934</y>
    </hint>
    <hint type="destinationlabel">
     <x>391</x>
     <y>-7</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>reset_prompt</sender>
   <signal>clicked()</signal>
   <receiver>standard_image_widget</receiver>
   <slot>reset_prompt_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>140</x>
     <y>684</y>
    </hint>
    <hint type="destinationlabel">
     <x>95</x>
     <y>-10</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>reset_negative_prompt</sender>
   <signal>clicked()</signal>
   <receiver>standard_image_widget</receiver>
   <slot>reset_negative_prompt_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>359</x>
     <y>681</y>
    </hint>
    <hint type="destinationlabel">
     <x>211</x>
     <y>-12</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>controlnet</sender>
   <signal>currentTextChanged(QString)</signal>
   <receiver>standard_image_widget</receiver>
   <slot>handle_controlnet_changed(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>136</x>
     <y>771</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>25</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>prompt</sender>
   <signal>textChanged()</signal>
   <receiver>standard_image_widget</receiver>
   <slot>prompt_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>84</x>
     <y>624</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>12</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>negative_prompt</sender>
   <signal>textChanged()</signal>
   <receiver>standard_image_widget</receiver>
   <slot>negative_prompt_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>362</x>
     <y>631</y>
    </hint>
    <hint type="destinationlabel">
     <x>-3</x>
     <y>595</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>upscale_model</sender>
   <signal>currentTextChanged(QString)</signal>
   <receiver>standard_image_widget</receiver>
   <slot>upscale_model_changed(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>114</x>
     <y>881</y>
    </hint>
    <hint type="destinationlabel">
     <x>1</x>
     <y>108</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>face_enhance</sender>
   <signal>toggled(bool)</signal>
   <receiver>standard_image_widget</receiver>
   <slot>face_enhance_toggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>60</x>
     <y>910</y>
    </hint>
    <hint type="destinationlabel">
     <x>1</x>
     <y>381</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>image_to_canvas()</slot>
  <slot>similar_image()</slot>
  <slot>variant_image()</slot>
  <slot>delete_image()</slot>
  <slot>similar_batch()</slot>
  <slot>confirm_delete()</slot>
  <slot>cancel_delete()</slot>
  <slot>export_image()</slot>
  <slot>upscale_2x_clicked()</slot>
  <slot>reset_prompt_clicked()</slot>
  <slot>reset_negative_prompt_clicked()</slot>
  <slot>handle_controlnet_changed(QString)</slot>
  <slot>prompt_changed()</slot>
  <slot>negative_prompt_changed()</slot>
  <slot>upscale_model_changed(QString)</slot>
  <slot>face_enhance_toggled(bool)</slot>
 </slots>
</ui>
